src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -317,10 +318,9 @@ export class EvaluationPlugin extends UIPlugin {
   getCorrespondingFormulaCell(position: CellPosition): FormulaCell | undefined {
     const cell = this.getters.getCell(position);
 
-    if (cell && cell.content) {
-      if (cell.isFormula && !isBadExpression(cell.content)) {
-        return cell;
-      }
+    if (cell && cell.isFormula) {
+      return isBadExpression(cell.compiledFormula.tokens) ? undefined : cell;
+    } else if (cell && cell.content) {
```

Comment:
could be `cell?.isFormula` and `cell?.content`

Reply:
I think cell && ... is easier to read, no ?

Reply:
I don't care. It's nitpicking